src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -395,11 +457,23 @@ public static BigDecimal constructDecimal(IonStruct constraintStruct) throws Exc
     }
 
     /**
-     * Generate random integers which composed by different length of data into the output file.
+     * Generate random integers which is conformed with the constraints provided in ISL.
+     * @param constraintStruct is an IonStruct which contains the top-level constraints in Ion Schema.
+     * @return constructed integers
+     * @throws Exception if error occurs when parsing the constraint 'valid_values'.
      */
-    public static long constructInt() {
+    public static long constructInt(IonStruct constraintStruct) throws Exception {
         Random random = new Random();
-        long longValue = random.nextLong();
+        long longValue;
+        IonList range = IonSchemaUtilities.getConstraintValueAsRange(constraintStruct, IonSchemaUtilities.KEYWORD_VALID_VALUES);
+        if (range != null) {
+            // Convert IonValue to long
+            long lowerBound = Long.valueOf(range.get(0).toString());
+            long upperBound = Long.valueOf(range.get(1).toString());
+            longValue = lowerBound + (long)(Math.random() * (upperBound - lowerBound));
```

Comment:
You already have a `Random` instance, and should probably use that here instead of `Math.random()` (which creates its own `Random` instance). That should allow you to use `nextLong()` to avoid the cast.

Reply:
> You already have a `Random` instance, and should probably use that here instead of `Math.random()` (which creates its own `Random` instance). That should allow you to use `nextLong()` to avoid the cast.

To avoid this situation, the new updates in the following commit use `ThreadLocalRandom.current().nextLong(lowerBound, upperBound) ` to generate the random value which allows to set range.